Add integration test for non-eksctl created clusters

aclevername opened the pull request:
### Description
This PR introduces a set of integration test to ensure that the non-eksctl created clusters are supported
### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [ ] Added labels for change area (e.g. `area/nodegroup`), target version (e.g. `version/0.12.0`) and kind (e.g. `kind/improvement`)
- [ ] Make sure the title of the PR is a good description that can go into the release notes

aclevername commented on integration/main.go:
```diff
@@ -90,6 +90,8 @@ func listModules() []string {
 			dirs = append(dirs, path.Join(testsDir, f.Name()))
 		}
 	}
+
+	dirs = []string{path.Join(testsDir, "unowned_cluster")}
```
TODO revert 😅

aclevername commented:
TODO: update docs

aclevername commented:
The test takes 70mins due to the create & upgrade cluster length 😢 
```
[0] • [SLOW TEST:4235.269 seconds]
[0] (Integration) [non-eksctl cluster & nodegroup support]
[0] /Users/jake/weave/eksctl/integration/tests/unowned_cluster/unowned_cluster_test.go:36
[0]   Get, upgrade & delete cluster/nodegroups
[0]   /Users/jake/weave/eksctl/integration/tests/unowned_cluster/unowned_cluster_test.go:37
[0]     should work
[0]     /Users/jake/weave/eksctl/integration/tests/unowned_cluster/unowned_cluster_test.go:50
[0] ------------------------------
[0]
[0] JUnit report was created: /Users/jake/weave/eksctl/test-results/__Users__jake__weave__eksctl__integration__tests__unowned__cluster__unowned__cluster__test__go_1.xml
[0]
[0] Ran 1 of 1 Specs in 4235.270 seconds
[0] SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
[0] --- PASS: TestE2E (4235.27s)
[0] PASS
[0]
[0] Ginkgo ran 1 suite in 1h10m47.6510235s
[0] Test Suite Passed
[0]=================================== Done ===================================
2021/01/11 16:15:15 ================== Summary ================================
[0] Ran 1 of 1 Specs in 4235.270 seconds
[0] SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
[0] --- PASS: TestE2E (4235.27s)
[0] PASS
[0]
[0] Ginkgo ran 1 suite in 1h10m47.6510235s
[0] Test Suite Passed
2021/01/11 16:15:15 ================== See ya Space Cowboy! ===================
(1.19 → origin ✓) eksctl
```

Callisto13 reviewed (approved):
> The test takes 70mins 

😭

michaelbeaumont commented on CONTRIBUTING.md:
````diff
@@ -371,6 +371,13 @@ There are several ways to run the tests. Requirements are:
 
+The integration tests require the following to be set:
+```BASH
+export IT_SUBNET_1 # Subnet used to create manual eks cluster
````
I could also imagine instead passing in the VPC ID and just taking 2 subnets, could be simpler at least from a configuration point of view.

aclevername replied:
Yeah that would make it simper, but I don't think we currently have anyway of querying the VPC api so we'd have to setup a new client etc so not sure if its worth it. I was thinking about removing the env vars and setting up a CF template to create the network/roles but I don't know if its worth it either 🤷‍♂️

michaelbeaumont replied:
yeah it's definitely fine as is

michaelbeaumont replied:
i think it's worth using CFN to create the necessary infra though, instead of clicking it together so no one knows the specifics of the prerequisites

aclevername commented:
After merging I need to update our CI to ensure they have the new env vars

michaelbeaumont commented:
@aclevername https://app.circleci.com/settings/project/github/weaveworks/eksctl/environment-variables

aclevername commented:
Removed the env vars. I've introduced a new CF template that we use to create the subnets & roles. I used [example VPC CF template from AWS](https://docs.aws.amazon.com/eks/latest/userguide/create-public-private-vpc.html#vpc-public-only) and added in the roles. There might be extra parts to the CF template that are redundant (I don't have 100% understanding of all the components of it)

aclevername commented:
```
• [SLOW TEST:4300.089 seconds]
(Integration) [non-eksctl cluster & nodegroup support]
/Users/jake/weave/eksctl/integration/tests/unowned_cluster/unowned_cluster_test.go:41
  Get, upgrade & delete cluster/nodegroups
  /Users/jake/weave/eksctl/integration/tests/unowned_cluster/unowned_cluster_test.go:42
    should work
    /Users/jake/weave/eksctl/integration/tests/unowned_cluster/unowned_cluster_test.go:67
------------------------------

Ran 1 of 1 Specs in 4300.089 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
